bread/layout/components/datatable.py

```diff
@@ -257,15 +269,28 @@ def with_toolbar(
                 ),
                 hg.DIV(
                     hg.DIV(
-                        Search(
-                            widgetattributes={"autofocus": True},
-                            backend=search_backend,
-                            resultcontainerid=resultcontainerid,
-                            show_result_container=False,
+                        hg.FORM(
+                            searchbar,
+                            hg.Iterator(
+                                hg.C("request").GET.items(),
+                                "queryitem",
+                                hg.If(
+                                    hg.F(
+                                        lambda c: c["queryitem"][0]
+                                        != search_urlparameter
+                                    ),
+                                    hg.INPUT(
+                                        type="hidden",
+                                        name=hg.C("queryitem")[0],
+                                        value=hg.C("queryitem")[1],
+                                    ),
+                                ),
+                            ),
+                            method="GET",
                         ),
                         _class="bx--toolbar-search-container-persistent",
                     )
-                    if search_backend
+                    if search_urlparameter is not None
```

Comment:
maybe replace this with `hg.If` (the ternary operator in Python has earned many awards for being one of the ugliest features in the Language)

Reply:
I tend to use ```hg.If ```only if the condition could be lazy-evaluated. But I do not really see a sensible use case where ```search_urlparameter``` would need to be lazy. Therefore relying on Pythons conditional operator makes more sense here I think.

Right, I read about criticism a few times too .But I wonder if there is more to it than personal opinion. People who are used to the C-style ternary operator ```? :``` are likely to find this way of ordering a conditional operator confusing, but I find it actually quite readable.

Reply:
I guess the problem originates from the order being different in the if-else statements and in the if-else expressions. That's what makes it unintuitive for many people.